src/relax/transform/canonicalize_bindings.cc

```diff
@@ -247,7 +247,8 @@ class BindingCanonicalizer : public ExprMutator {
   }
 
   Expr VisitExpr_(const VarNode* var) override {
-    Var new_var = GetRef<Var>(var);
```

Comment:
Was there any bug you encountered due to not doing this visit before? If so, that would be a good regression test to throw in.

Reply:
Hmm, this was definitely required during initial development, and was intended to handle cases where a replacement in `ExprMutator::var_remap_` occurs outside of the planned replacements in the `ReplacementPlan.replace_usage_`.  This can happen if the removal of trivial bindings triggers `InferStructInfo` in a downstream operation.  However, that doesn't reproduce the bug, nor does that bug occur in anything on my local dev branch.

I'm removing this line from the PR, and can make a separate PR (with test case!) if/when the bug resurfaces.

Reply:
I don't think it's necessary to remove, especially since it seems like a correct change. I'm glad that there wasn't a bug that you encountered because of it.